torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -536,41 +545,55 @@ void validateAlignedVectorizeExtents(
       info.word_size);
 }
 
-bool validateAlignedVectorizedFusionInputOutput(
+void validateAlignedVectorizedFusionInputOutput(
     const IValue& aten_val,
-    int word_size) {
+    int word_size,
+    TensorView* tv) {
   TORCH_INTERNAL_ASSERT(aten_val.isTensor());
 
   const auto& aten_tensor = aten_val.toTensor();
 
-  if (reinterpret_cast<size_t>(aten_tensor.data_ptr()) %
-          (word_size * aten_tensor.dtype().itemsize()) !=
-      0) {
-    return false;
-  }
+  TORCH_INTERNAL_ASSERT(
+      reinterpret_cast<size_t>(aten_tensor.data_ptr()) %
+              (word_size * aten_tensor.dtype().itemsize()) ==
+          0,
+      "Vectorization of ",
+      tv->toString(),
+      " not possible as the memory address is not aligned. ",
+      "Address: ",
+      aten_tensor.data_ptr(),
+      ", vector word size: ",
+      word_size,
+      ", data type: ",
+      aten_tensor.dtype());
 
   // Traverse strides from the right-most domains. The rightmost
   // domain must have stride 1.
   int64_t cur_contig_stride = 1;
   bool still_rightmost = true;
   for (auto i = aten_tensor.ndimension() - 1; i >= 0; --i) {
-    auto stride = aten_tensor.strides().at(i);
-    auto size = aten_tensor.sizes().at(i);
+    const auto stride = aten_tensor.strides().at(i);
     // If this domain is contiguous, then not necessary to check the
     // stride. Otherwise, stride must be 1 if it's rightmost or
     // divisible by word_size.
```

Comment:
Isn't this only true if we merge all contiguous dimensions? Shouldn't this take into consideration the root domains used in the contig_id found from the vectorized id?

Reply:
Hmm, I thought these checks here are sufficient given that we also validate extents with `validateAlignedVectorizeExtents`. Do you have any specific example in your mind?